crates/rpc/rpc-builder/src/auth.rs

```diff
@@ -454,3 +449,72 @@ impl AuthServerHandle {
         self.ipc_endpoint.clone()
     }
 }
+
+/// A layer that adds a new JWT token to every request using JwtSecretService.
+#[derive(Debug, Clone)]
+struct JwtSecretLayer {
+    secret: JwtSecret,
+}
```

Comment:
can we move this to /src/layers ?

looks like the layers mod is not very clean because it mixes jwt types and layer. we could restructure this in a followup

Reply:
done, thank you! renamed it to `AuthClientLayer` for consistency with `AuthLayer`.